cfgov/v1/management/commands/report_raw_html.py

```diff
@@ -0,0 +1,142 @@
+import collections
+import csv
+import logging
+import re
+
+from django.apps import apps
+from django.core.management.base import BaseCommand
+
+from wagtail.fields import StreamField
+from wagtail.models import Site
+
+from v1.models import CFGOVPage
+
+
+logger = logging.getLogger(__name__)
+
+
+def find_pattern(data, pattern, key, path=None):
+    """Return the path and match for a given regular expression in JSON
+
+    For the given JSON-like data descend through the data structures looking
+    for strings that match the `re.compile()`ed pattern.
+
+    When one is found, the path to the match will be yielded along with any
+    matching strings. If the path goes through a a JSON array, it will include
+    the index in the array, and if it goes through a JSON object, it'll include
+    the a property of the object with the given key."""
+    if path is None:
+        path = []
+
+    # If it's a string, try to match it
+    if isinstance(data, str):
+        matches = pattern.findall(data)
+        if len(matches) > 0:
+            yield path, matches
+
+    # If its a mapping, iterate over its key/value pairs
+    elif isinstance(data, collections.Mapping):
+        local_path = path
+        if key in data:
+            local_path = path + [data[key]]
+
+        for item in data:
+            yield from find_pattern(data[item], pattern, key, path=local_path)
+
+    # If its a sequence, iterate over its members
+    elif isinstance(data, collections.Sequence) and not isinstance(data, str):
+        for index, item in enumerate(data):
+            local_path = path + [str(index)]
+            yield from find_pattern(item, pattern, key, path=local_path)
+
+
+class Command(BaseCommand):
+    help = (
+        "Discover raw HTML tags within &lt; &gt; entities in page fields. "
+        "Pass a list in the form of app_name.page_type.field for each page "
+        "type and field you want to report on."
+    )
+
+    # Match "&lt;" followed by any 0 or more characters that are not "&gt;",
+    # followed by "&gt;" This should match HTML that's encoded in HTML entities
+    # in the given fields, but it would be good to verify reported fields before
+    # assuming to much
+    html_tag_entity_pattern = r"&lt;[a-zA-Z]+(?!&gt;).*?&gt;"
+    html_tag_entity_re = re.compile(
+        html_tag_entity_pattern, re.MULTILINE | re.DOTALL
+    )
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "pagetype",
+            nargs="*",
+            help=(
+                "Specify the page type(s) and field to check."
+                "This should be given in the form app_name.page_type.field "
+                "to include a page type in the given app with the given field."
+                "For example, v1.BrowsePage.content. "
+                "If no page types are given, -f/--field is required."
```

Comment:
```suggestion
                "For example, v1.BrowsePage.content."
```

Does this refer to an option from a previous iteration of this script?

Reply:
Yes indeed.